pilot/pkg/config/kube/gateway/deploymentcontroller.go

```diff
@@ -323,12 +324,6 @@ func (d *DeploymentController) configureIstioGateway(log *istiolog.Scope, gw gat
 	}
 	log.Info("reconciling")
 
-	var ns *corev1.Namespace
-	if d.namespaces != nil {
-		ns = d.namespaces.Get(gw.Namespace, "")
-	}
-	proxyUID, proxyGID := inject.GetProxyIDs(ns)
```

Comment:
This change breaks Gateway API in cluster-wide mode, e.g. gateway controller.

Reply:
+1 I think this code shouldn't be removed but only surrounded by an if-clause that checks whether we're running in multi-tenant mode. In cluster-wide, behavior must match upstream ie this code block must be executed as usual